### ISSUE TITLE
Program running reported null pointer exception, resulting in stack overflow

### ISSUE THREAD

**Issue by Truly-Zheng** [springdoc/springdoc-openapi#2388]
![image](https://github.com/springdoc/springdoc-openapi/assets/56818082/34da9976-3982-4205-b471-7d84ebe74dd7)
![image](https://github.com/springdoc/springdoc-openapi/assets/56818082/0f2d59b7-9f97-4f34-ae77-d0caba6753c2)

My entity class extend the com.baomidou.mybatisplus.extension.activerecord.Model class, the runtime found a null pointer exception, this Model class uses generic references.

![image](https://github.com/springdoc/springdoc-openapi/assets/56818082/dd98e782-93b5-4f2b-8894-16c5551fdef3)

When I remove the extends class, the program works fine.hope to resolve the problem soon.
thanks a lot.

**Comment by marcospds** [comment]
Good evening, I was unable to simulate this problem, could you provide an example project of the error?
I tried it here: https://github.com/marcospds/springdoc-stack-overflow

**Comment by uc4w6c** [comment]
This ticket will be closed and can be reopened if the relevant information are provided.

**Comment by Truly-Zheng** [comment]
Hello, you added this configuration 
springdoc.default-flat-param-object=true

The above error will occur.

**Comment by marcospds** [comment]
Good evening @Truly-Zheng, I couldn't replicate your issue. I updated my project with the parameter you mentioned, and I also made some changes to make it more similar to your screenshots, but the error still doesn't occur. Another thing I noticed is that your SysUser class doesn't have the @Entity and @Id annotations, your project must have some other peculiarity causing the problem. Could you provide your project as an example or create a fork of the [sample](https://github.com/marcospds/springdoc-stack-overflow) to show how your project is set up so that we can replicate the issue?

**Comment by Truly-Zheng** [comment]
Hello,good morning @Marcos Paulo dos Santos
I have a demo here that you can refer to. Please refer to the attachment for details.




zhengcl
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "springdoc/springdoc-openapi"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月12日(星期四) 中午11:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [springdoc/springdoc-openapi] Program running reported null pointer exception, resulting in stack overflow (Issue #2388)





 
Good evening @Truly-Zheng, I couldn't replicate your issue. I updated my project with the parameter you mentioned, and I also made some changes to make it more similar to your screenshots, but the error still doesn't occur. Another thing I noticed is that your SysUser class doesn't have the @entity and @id annotations, your project must have some other peculiarity causing the problem. Could you provide your project as an example or create a fork of the sample to show how your project is set up so that we can replicate the issue?

**Comment by Truly-Zheng** [comment]
Hello，good morning @marcospds 
I have a demo here that you can refer to. 
https://github.com/Truly-Zheng/springdoc-stack-overflow.git

**Comment by marcospds** [comment]
Hello @Truly-Zheng, your project is using springdoc-openapi 1.7 which is no longer supported, @uc4w6c could confirm?

Disclaimer:
```
springdoc-openapi v1.7.0 is the latest Open Source release supporting Spring Boot 2.x and 1.x.
An extended support for springdoc-openapi v1 project is now available for organizations that need support beyond 2023.
For more details, feel free to reach out: sales@springdoc.org
```

**Comment by uc4w6c** [comment]
@marcospds 
Thank you.
But if v2.2.0 occur issue too, after we fix it, we release v2.2.1 or v2.3.0.

@Truly-Zheng 
This issue was already fixed it https://github.com/springdoc/springdoc-openapi/pull/2291.
But v1 is not supported. Could you use v2.
Alternatively, by separating the Bean that receives data through RequestBody from the Bean responsible for database operations, you can resolve the issue.
